src/darwin/Framework/CHIP/MTROperationalBrowser.h

```diff
@@ -0,0 +1,47 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDeviceControllerFactory.h>
+#import <dns_sd.h>
+
+class MTROperationalBrowser
```

Comment:
would it make sense to have a core object for these things? MinMDNS can have similar capabiltity as it receives every single mdns packet.

Reply:
> would it make sense to have a core object for these things? 

Perhaps, but the core "browse" API surface we have does not support the things that need to be done here.  MinMDNS can do it, the Darwin backend can do it, dunno about the other backends, but the API on top of them cannot.

If we fix that, then we can change the bits here to use that core object instead.  But it's going to take some API refactoring to make it happen.

Reply:
My concern with the platform-only approach is that it makes it more likely that functionality diverges across platforms, so developing apps on one platform vs another starts becoming different in terms of functionality.

I believe it is fine platforms to have code that is platform-specific, however I would also argue in that case it should be part of a separate repo not directly in the CHIP SDK (like we have embedded platform SDKs for various targets which provide different functionality)

Reply:
We have tons of differences across platforms right now.  Darwin and Android have different API surfaces with different capabilities, for example.  And neither one matches the API surface of the raw C++ SDK.

Note that in this case it's not even an API difference.  It's just an implementation detail of the Darwin framework.  Just like the fact that we auto-resubscribe subscriptions by default is an implementation detail, and the fact that we ensure that we don't blow out read handler limits is a detail, etc.

I agree that ideally things that can be done in the core should be done in the core.  In this case, I tried to do this in the core, and it's not possible there without significantly more API work than I have time for right now.  That's not sufficient reason to not ship the functionality on iOS, though...  And if I find more time, or if someone else makes the core more reasonable, we can move the functionality into the core.

The argument about same vs separate repo is an interesting one, but unless you are arguing that Matter.framework should not be in-tree at all, I don't think it really applies in this case.